Code/src/main/java/nl/utwente/group10/ui/components/anchors/ConnectionAnchor.java

```diff
@@ -48,55 +48,115 @@ public ConnectionAnchor(Block block, CustomUIPane pane) {
-     *         that connection is fully connected.
+     * @return True if the primary connection is connected.
+     */
+    public boolean isPrimaryConnected() {
+        return isConnected(0);
+    }
+    
+    /**
+     * @param index Index of the connection to check
+     * @return Wether or not the connection specified by the index is connected.
      */
-    public boolean isConnected() {
-        return hasConnection() && getConnection().get().isConnected();
+    public boolean isConnected(int index){
+        return index >= 0 && index < getConnections().size() && getConnections().get(index).isConnected();
     }
 
-    /** Returns true if this anchor can connect to a connection. */
-    public abstract boolean canConnect();
+    /** Wether or not this anchor allows adding an extra connection. */
+    public abstract boolean canAddConnection();
 
     /**
-     * This method provides a shortcut to get the anchor on the other side of the Connection from this anchor.
-     * @return Optional.empty if this anchor does not have a Connection, or that Connection is not connected on both sides.
-     *         Else it returns an Optional containing the other anchor.
+     * This method provides a shortcut to get the anchors on the other side of the Connection from this anchor.
+     * @return A list of each potential opposite anchor for each Connection this anchor has.
      */
-    public Optional<? extends ConnectionAnchor> getOtherAnchor() {
-        if (hasConnection()) {
-            Optional<OutputAnchor> out = getConnection().get()
-                    .getOutputAnchor();
-            Optional<InputAnchor> in = getConnection().get().getInputAnchor();
-            if (out.isPresent() && out.get().equals(this)) {
-                return in;
-            } else if (in.isPresent() && in.get().equals(this)) {
-                return out;
+    public List<Optional<? extends ConnectionAnchor>> getOppositeAnchors() {
+        List<Optional<? extends ConnectionAnchor>> list = new ArrayList<Optional<? extends ConnectionAnchor>>();
+        for(Connection c : getConnections()){
+            if(c.isConnected()){
+                if(c.getInputAnchor().isPresent() && c.getInputAnchor().get().equals(this)){
+                    list.add(c.getOutputAnchor());
+                }else if(c.getOutputAnchor().isPresent() && c.getOutputAnchor().get().equals(this)){
+                    list.add(c.getInputAnchor());
+                }else{
+                    list.add(Optional.empty());
+                }
+            }else{
+                list.add(Optional.empty());
             }
         }
-        return Optional.empty();
+        return list;
+    }
+    
+    /**
+     * @return Just the primary's opposite anchor.
+     */
+    public Optional<? extends ConnectionAnchor> getPrimaryOppositeAnchor(){
+        if(getOppositeAnchors().size()>0){
+            return getOppositeAnchors().get(0);
+        }else{
+            return Optional.empty();
```

Comment:
Misschien heb je iets aan `Optional.map`, die een functie `f(x)` toepast op een `Optional<x>` als die present is en anders een lege optional teruggeeft. Zou kunnen dat je daarmee de `else return Optional.empty();` zou kunnen vervangen.